src/main/kotlin/com/demonwav/mcdev/platform/mixin/action/GenerateAccessorHandler.kt

```diff
@@ -0,0 +1,496 @@
+
+        val accessors = arrayListOf<PsiMethod>()
+
+        if (generateGetter) {
+            val prefix = if (target.type == PsiType.BOOLEAN) "is" else "get"
+            val method = factory.createMethodFromText(
+                """
+                @${MixinConstants.Annotations.ACCESSOR}
+                ${staticPrefix(isStatic)}ReturnType $prefix${target.name.capitalize()}()${methodBody(isStatic)}
+                """.trimIndent(),
+                mixin
+            )
+            target.typeElement?.let { method.returnTypeElement?.replace(it) }
+            accessors.add(method)
+        }
+        if (generateSetter) {
+            val method = factory.createMethodFromText(
+                """
+                @${MixinConstants.Annotations.ACCESSOR}
+                ${staticPrefix(isStatic)}void set${target.name.capitalize()}(ParamType ${target.name})${methodBody(isStatic)}
+                """.trimIndent(),
+                mixin
+            )
+            target.typeElement?.let { method.parameterList.parameters[0].typeElement?.replace(it) }
+            if (target.modifierList?.hasExplicitModifier(PsiModifier.FINAL) == true) {
+                AddAnnotationFix(MixinConstants.Annotations.MUTABLE, method).applyFix()
+            }
+            accessors.add(method)
+        }
+
+        return accessors
+    }
+
+    private fun generateInvoker(project: Project, target: PsiMethod, mixin: PsiClass): PsiMethod? {
+        val factory = JavaPsiFacade.getElementFactory(project)
+
+        val isStatic = target.modifierList.hasExplicitModifier(PsiModifier.STATIC) || target.isConstructor
+        val name = if (target.isConstructor) {
+            "create${target.containingClass?.name?.capitalize()}"
+        } else {
+            "call${target.name.capitalize()}"
+        }
+
+        val method = factory.createMethodFromText(
+            """
+            @${MixinConstants.Annotations.INVOKER}
+            ${staticPrefix(isStatic)}ReturnType $name()${methodBody(isStatic)}
+            """.trimIndent(),
+            mixin
+        )
+        if (target.isConstructor) {
+            val targetClass = target.containingClass ?: return null
+            method.returnTypeElement?.replace(factory.createTypeElement(factory.createType(targetClass)))
+        } else {
+            target.returnTypeElement?.let { method.returnTypeElement?.replace(it) }
+        }
+        method.parameterList.replace(target.parameterList)
+        method.throwsList.replace(target.throwsList)
+
+        return method
+    }
+
+    private fun staticPrefix(isStatic: Boolean): String {
+        return if (isStatic) {
+            "static "
+        } else {
+            ""
+        }
+    }
+
+    private fun methodBody(isStatic: Boolean): String {
+        return if (isStatic) {
+            "{ throw new java.lang.UnsupportedOperationException(); }"
+        } else {
+            ";"
+        }
+    }
+
+    private class HeaderPanel : JComponent() {
```

Comment:
Not a requirement or anything but this could be implemented using the Kotlin UI DSL for IntelliJ - something I would like to move my UIs to if feasible, but I don't have much experience with it so we'll see.

This would look like this probably:

```kt
private class Header {
    val gettersCheckbox = JBCheckBox("Generate Getter Accessors", true)
    val settersCheckbox = JBCheckBox("Generate Setter Accessors")
    val panel: JComponent = panel {
        row { gettersCheckbox() }
        row { settersCheckbox() }
    }

    init {
        gettersCheckbox.addItemListener(swap(settersCheckbox))
        settersCheckbox.addItemListener(swap(gettersCheckbox))
    }

    private fun swap(other: JBCheckBox) = ItemListener {
        if (it.stateChange == ItemEvent.DESELECTED) {
            other.isSelected = true
        }
    }
}
```

In my test at least the result seems identical.

Reply:
For me it seems like there is a bit cut off the top when I do this, it's weird
![image](https://user-images.githubusercontent.com/13084089/78980323-94406800-7b15-11ea-94b7-fc2ccde42f03.png)

Reply:
Oh yeah you're right, that's strange. Maybe their UI DSL isn't so easy to use then.